src/lib/validation.ts

```diff
@@ -0,0 +1,432 @@
+		if (!isValidShortText(key)) {
+			console.log('debug: Non-valid short text label key passed to validation.isValidLabelsObject');
+			console.log(`\tkey: ${inspect(key)}`);
+			return false;
+		}
+
+		if (!LABEL_NAME_REGEX.test(key)) {
+			console.log('debug: Invalid label name passed to validation.isValidLabelsObject');
+			console.log(`\tkey: ${inspect(key)}`);
+			return false;
+		}
+
+		if (!_.isString(val)) {
+			console.log('debug: Non-string value passed to validation.isValidLabelsObject');
+			console.log(`\tval: ${inspect(val)}`);
+			return false;
+		}
+
+		return true;
+	});
+}
+
+function undefinedOrValidEnv(val: EnvVarObject): boolean {
+	return !(val != null && !isValidEnv(val));
+}
+
+/**
+ * isValidDependentAppsObject
+ *
+ * Given a dependent apps object from a state endpoint, validate it
+ *
+ * TODO: Type the input
+ */
+export function isValidDependentAppsObject(apps: any): boolean {
+	if (!_.isObject(apps)) {
+		console.log('debug: non-object passed to validation.isValidDependentAppsObject');
+		console.log(`\tapps: ${inspect(apps)}`);
+		return false;
+	}
+
+	return _.every(apps, (val, appId) => {
+		val = _.defaults(_.clone(val), {
+			config: undefined,
+			environment: undefined,
+			commit: undefined,
+			image: undefined,
+		});
+
+		if (!(isValidShortText(appId) && checkInt(appId))) {
+			console.log('debug: Invalid appId passed to validation.isValidDependentAppsObject');
+			console.log(`\tappId: ${inspect(appId)}`);
+			return false;
+		}
+
+		return _.conformsTo(val, {
+			name: (n: any) => {
+				if (!isValidShortText(n)) {
+					console.log('debug: Invalid name passed to validation.isValidDependentAppsObject');
+					console.log(`\tname: ${inspect(n)}`);
+					return false;
+				}
+				return true;
+			},
+			image: (i: any) => {
+				if (val.commit != null && !isValidShortText(i)) {
+					console.log('debug: non valid image passed to validation.isValidDependentAppsObject');
+					console.log(`\timage: ${inspect(i)}`);
+					return false;
+				}
+				return true;
+			},
+			commit: (c: any) => {
+				if (c != null && !isValidShortText(c)) {
+					console.log('debug: invalid commit passed to validation.isValidDependentAppsObject');
+					console.log(`\tcommit: ${inspect(c)}`);
+					return false;
+				}
+				return true;
+			},
+			config: (c: any) => {
+				if (!undefinedOrValidEnv(c)) {
+					console.log('debug; Invalid config passed to validation.isValidDependentAppsObject');
+					console.log(`\tconfig: ${inspect(c)}`);
+					return false;
+				}
+				return true;
+			},
+			environment: (e: any) => {
+				if (!undefinedOrValidEnv(e)) {
+					console.log('debug; Invalid environment passed to validation.isValidDependentAppsObject');
+					console.log(`\tenvironment: ${inspect(e)}`);
+					return false;
+				}
+				return true;
+			}
+		});
+	});
+}
+
+function isValidService(service: any, serviceId: string): boolean {
+	if (!(isValidShortText(serviceId) && checkInt(serviceId))) {
```

Comment:
Can you refactor this to `if (!isValidShortText(serviceId) || !checkInt(serviceId)) {` ?